backend/src/stats.rs

```diff
@@ -154,27 +175,23 @@ impl BlockStats {
             }
             None => {
                 debug!("Could not identify pool at height {}", height);
-                UNKNOW_POOL_ID
+                UNKNOWN_POOL_ID
             }
         };
 
-        BlockStats {
+        Ok(BlockStats {
             height: height,
             date: date.to_string(),
-            version: block.header.version.to_consensus(),
-            nonce: block.header.nonce as i32,
-            bits: block.header.bits.to_consensus() as i32,
+            version: block.version.to_consensus(),
+            nonce: block.nonce as i32,
+            bits: i32::from_str_radix(&block.bits, 16)?,
 
             pool_id,
 
-            size: block.total_size() as i64,
-            stripped_size: block
-                .txdata
-                .iter()
-                .map(Transaction::base_size)
-                .sum::<usize>() as i64,
-            vsize: block.txdata.iter().map(Transaction::vsize).sum::<usize>() as i64,
-            weight: block.weight().to_wu() as i64,
+            size: block.size,
+            stripped_size: block.stripped_size,
```

Comment:
While working on #30, I noticed that this actually fixes a bug in the old `stripped_size` calculation.

The old one is missing the 80 header bytes and the 1 or 3 bytes for the block transaction count.